# Variables
#### Variables are containers that store data in a Python program.
#### Variables can store different types of data types such as string, integer, float, boolean.

In [1]:
# creating a string variable
can = 'soda'

In [2]:
can

'soda'

In [3]:
# changing the value of the variable
can = 'beans'

In [4]:
can

'beans'

In [5]:
print(can)

beans


In [6]:
# data type
type(can)

str

In [7]:
# creating an integer variable (price)
pizza = 100
pizza

100

In [8]:
# creating a float variable (price)
samosa = 20.5
ice_cream = 100

In [9]:
# making addition
total = pizza + samosa + ice_cream
total

220.5

In [10]:
# data type
type(total)

float

In [11]:
# creating boolean value
above_thresold = total > 200
above_thresold

True

In [12]:
# same variable data type can be changed directly
can = 'beans'
print(type(can))

can = 6.7
print(type(can))

<class 'str'>
<class 'float'>


In [13]:
# if the value of multiple variables are same, their id value will be same
foo = 'jalebi'
bar = foo

print(id(foo))
print(id(bar))

134709588184048
134709588184048


#### **Reserved keywords** and **Special Characters** cannot be used as variable name.

# Numbers

In [14]:
# Calculating area of a triangle
base = 10
height = 7

area = (base * height) / 2
area

35.0

In [15]:
# Data type
type(area)

float

In [16]:
x = 10
y = 3

# Floor division
print(x//y)

# Modulus operator
print(x%y)

3
1


In [17]:
# Exponentiation
x**y

1000

In [18]:
# e means "10 to the power"
2.5e4

25000.0

In [19]:
2.5e-3

0.0025

In [20]:
# round() will give the whole number
food = 120.34
rent = 500.4
utilities = 200.5

total = food + rent + utilities
print(round(total))

821


In [21]:
# custom data point of round() function
round(total, 2)

821.24

In [22]:
# string variables
food = 120.34
rent = 500.67

# converting strings to floats
total = float(food) + float(rent)
total

621.01

In [23]:
import math

x = 16
math.sqrt(x) # square root

4.0

In [24]:
math.floor(3.7) # floor value

3

In [25]:
math.ceil(3.7) # ceiling value

4

In [26]:
math.pi # pi value

3.141592653589793

In [27]:
format(5, 'b') # binary representation of the number

'101'

# Strings

#### In Python, strings are immutable, meaning they cannot be changed once created.

In [28]:
first = 'Sk Golam'
last = 'Mahmud'

name = first + ' ' + last
name

'Sk Golam Mahmud'

In [29]:
name = f"{first} {last}"
name

'Sk Golam Mahmud'

In [30]:
age = 28
info = f"{first} {last} is {age} years old"
info

'Sk Golam Mahmud is 28 years old'

In [31]:
# specific letter or words can be generated from string
print(name[0])
print(name[0:8])

S
Sk Golam


In [32]:
# triple quote for multiple lines
recipe = '''
Veg biryani with
saffron, cardamom, and cloves,
garnished with fried onions and cilantro.
'''

print(recipe)


Veg biryani with 
saffron, cardamom, and cloves, 
garnished with fried onions and cilantro.



In [33]:
# finding specific word in recipe
spice = "cardamom"
spice in recipe

True

In [34]:
"milk" in recipe

False

In [35]:
# in order to find a specific word's index
recipe.find("cardamom")

28

In [36]:
recipe.index("cardamom")

28

In [37]:
# replace
s = "Patient was charged 100$ for lab test"
s

'Patient was charged 100$ for lab test'

In [38]:
s = s.replace("100$", "10$")
s

'Patient was charged 10$ for lab test'

In [39]:
s.upper()

'PATIENT WAS CHARGED 10$ FOR LAB TEST'

In [40]:
s.lower()

'patient was charged 10$ for lab test'

In [41]:
# in order to find numeric value
"10w".isdigit()

False

In [42]:
# Joining integer as string
text = "My age is: "
age = 27

text + str(age)

'My age is: 27'

In [43]:
# split
tickers = "AAPL|NVDA|RIL|GOGL"
tickers.split("|")

['AAPL', 'NVDA', 'RIL', 'GOGL']

In [44]:
# strip
data = "    Hi, How are you?    "
data.strip()

'Hi, How are you?'

In [45]:
# detecting specific file
file_name = "report.pdf"
file_name.endswith(".pdf")

True

# Exercise

### Task 1

Create a variable named pi and store the value 22/7 in it. Now check the data type of this variable.

In [46]:
pi = 22/7
type(pi)

float

### Task 2

Create a variable called ```for``` and assign it a value 4. See what happens and find out the reason behind the behavior that you see.

In [47]:
# reserved keywords cannot be used

### Task 3

Below we have defined the variables for storing the principal amount, rate of interest and time. You need to calculate the simple interest for 3 years. Once simple interest is calculated, calculate the total amount you will have at the end of the tenure

In [48]:
principle_amount = 567.00
rate_of_interest = 5.6
time = 3

Now let's calculate the simple intrest using simple math formula:

`simple interest  = P x R x T / 100`

where:

    P = principle amount
    R = rate of interest
    T = time

In [49]:
# calculate simple interest
simple_interest = principle_amount * rate_of_interest * time / 100
simple_interest

95.25599999999999

In [50]:
# calculate total amount after including interest
total_amount = principle_amount + simple_interest
total_amount

662.256

### Task 4

There is a circular pond in a village. This pond has a radius of 84 meter. Can you find the area of the pond?

In [51]:
import math
area = math.pi * 84**2
area

22167.07776372958

### Task 5

If there is a 2000 liter water in a square meter, what is the total amount of water in this pond?

**Note:** For simplicity, use the value of pi as 3.14.

In [52]:
total_water = area * 2000
total_water

44334155.52745916

### Task 6

If you cross a 490-meter-long street in 7 minutes, then what is your speed in meter per seconds. Print your answer with only two decimal points

Hint:
        Speed = Distance / Time

In [53]:
distance = 490
time = 7 * 60

speed = distance / time
round(speed, 2)

1.17

### Task 7

Create two variables to store how many fruits and vegetables you eat in a day. The value should be numeric for example 3 fruits and 4 vegetables. Now Print "I eat x vegetables and y fruits daily" where x and y presents vegetables and fruits that you eat every day. Use python f string for this.

In [54]:
fruits = 3
vegetables = 4

print(f"I eat {vegetables} vegetables and {fruits} fruits daily")

I eat 4 vegetables and 3 fruits daily


### Task 8

Create a variable and store the string “The Himalayas are one of the youngest mountain range on the planet.”

1. Print ‘The Himalayas’ using slice operator
1. Print “mountain range” using negative index
1. Print “The Himalayas on the planet” using slice as well as f-string

In [55]:
sentence = "The Himalayas are one of the youngest mountain range on the planet."

# 1. Printing 'The Himalayas' using slice
print(sentence[0:13])

# 2. Printing 'mountain range' using negative index
print(sentence[-29:-15])

# 3. Printing 'The Himalayas on the planet' using slice and f-string
part1 = sentence[0:13]         # 'The Himalayas'
part2 = sentence[-15:-1]       # 'on the planet'
print(f"{part1}{part2}")


The Himalayas
mountain range
The Himalayas on the planet


### Task 9

You have created a string variable called string= ”There are 9 planets in the solar system”. After some time, you have realized that your sentence is incorrect and there are only 8 planets, now correct your sentence by replacing the incorrect words.

In [56]:
a = "There are 9 planets in the solar system"
a = a.replace("9", "8")
a

'There are 8 planets in the solar system'

### Task 10

Imagine you are a shop owner tracking sales of three products throughout the day. At the end of the day, you want to calculate the total sales from these products. Product quantity and prices are given below in the next code cell,

Task: Write a program that:

1. Calculates the total sales for each product.
1. Summarizes the total sales for the day.
1. Prints this summary in a formatted manner.

Expected Output:

```
Daily Sales Summary:
- Product 1: Sold 15 units at $20.0 each, Total: $300.0
- Product 2: Sold 10 units at $35.0 each, Total: $350.0
- Product 3: Sold 20 units at $12.0 each, Total: $240.0
Total Sales for the Day: $890.0
```

In [57]:
# Prices and quantities for each product
product1_price = 20.0  # Price per unit for product 1
product1_quantity = 15  # Quantity sold of product 1

product2_price = 35.0  # Price per unit for product 2
product2_quantity = 10  # Quantity sold of product 2

product3_price = 12.0  # Price per unit for product 3
product3_quantity = 20  # Quantity sold of product 3

In [58]:
# 1. total sales for each product
total_sales_product1 = product1_price * product1_quantity
total_sales_product2 = product2_price * product2_quantity
total_sales_product3 = product3_price * product3_quantity

In [59]:
# 2. total sales for the day
total_sales_day = total_sales_product1 + total_sales_product2 + total_sales_product3

In [60]:
# 3. summary in a formatted manner
print(f"Total sales for product 1: {total_sales_product1}")
print(f"Total sales for product 2: {total_sales_product2}")
print(f"Total sales for product 3: {total_sales_product3}")

print(f"Total sales for the day: {total_sales_day}")

Total sales for product 1: 300.0
Total sales for product 2: 350.0
Total sales for product 3: 240.0
Total sales for the day: 890.0
